src/components/EmailNotifications/EmailInput.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from 'react'
+import {
+  Field,
+  GU,
+  IconCheck,
+  IconCross,
+  TextInput,
+  textStyle,
+  useInside,
+  useTheme,
+} from '@aragon/ui'
+
+const EmailInput = React.memo(function EmailInput({
+  existingEmail,
+  status,
+  ...inputProps
+}) {
+  const theme = useTheme()
+  const [error, setError] = useState('')
+  const [insideModal] = useInside('NotificationsModal')
+
+  useEffect(() => {
+    if (status === 'invalid') {
```

Comment:
We most likely want to use `getDerivedStateFromProps` instead of an `useEffect()` for this.

However, rather than that, I've been thinking that this component is currently at an awkward position, abstraction-wise. I see two directions we could go in:

1. Remove `status` (see comments above, also couples this component deeply with `useInput()`) and `existingEmail`, and use `valid` and `value` as props instead. This removes as much "magic" from this component as possible.
2. Encapsulate `useInput()` and the validation functions into this component, so users don't have to worry about those, and they can just consume the `onChange` value. The adds as much "magic" into the component as possible.

I was originally thinking about 2, but after some thought, 1 may be the simpler and better option, as pushing validation into input components is not something we've standardized on yet.

cc @bpierre

Reply:
> We most likely want to use `getDerivedStateFromProps` instead of an `useEffect()` for this.

I think a good equivalent for function components would be `useMemo()`?

```jsx
const error = useMemo(() => {
  return status === 'invalid'? 'Please enter…' : ''
}, [status])
```

Or an expression directly:

```jsx
const error = status === 'invalid'? 'Please enter…' : ''
```

> I was originally thinking about 2, but after some thought, 1 may be the simpler and better option, as pushing validation into input components is not something we've standardized on yet.

Agreed, 1 seems reasonable as a first step. Having 2 would be nice, but there are many ways we can do it, and we might want to do it at the `TextInput` level first.

Reply:
so to sumarize i shouldn't return any status from here?